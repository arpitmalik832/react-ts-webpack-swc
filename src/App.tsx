import { RouterProvider } from 'react-router-dom';

import router from './routes';
import useAppMount from './hooks/useAppMount';

function App() {
  useAppMount();

  return <RouterProvider router={router} />;
}

export default App;
