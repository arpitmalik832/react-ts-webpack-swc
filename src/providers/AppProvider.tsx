import { ReduxProvider } from '@arpitmalik832/react-js-rollup-monorepo-library';

import App from '../App';
import store from '../redux/store/store';

function AppProvider() {
  return (
    <ReduxProvider store={store}>
      <App />
    </ReduxProvider>
  );
}

export default AppProvider;
