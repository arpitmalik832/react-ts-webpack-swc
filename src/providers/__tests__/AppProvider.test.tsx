/**
 * AppProvider unit tests.
 * @file This file is saved as `AppProvider.test.jsx`.
 */
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ReduxProvider } from '@arpitmalik832/react-ts-rollup-library';

import AppProvider from '../AppProvider';
import App from '../../App';

// Mock the dependencies
jest.mock('../../App', () => ({
  __esModule: true,
  default: jest.fn(() => <div data-testid="mock-app">Mock App</div>),
}));

jest.mock('../../redux/store/store', () => ({
  __esModule: true,
  default: {
    getState: jest.fn(),
    dispatch: jest.fn(),
  },
}));

jest.mock('@arpitmalik832/react-ts-rollup-library', () => ({
  __esModule: true,
  ReduxProvider: jest.fn(({ children }) => (
    <div data-testid="mock-redux-wrapper">{children}</div>
  )),
}));

describe('AppProvider Tests', () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it('renders without crashing', () => {
    const { container } = render(<AppProvider />);
    expect(container).toBeTruthy();
  });

  it('renders ReduxProvider with correct props', () => {
    render(<AppProvider />);

    // Check if ReduxProvider was called with correct store prop
    expect(ReduxProvider).toHaveBeenCalledTimes(1);
  });

  it('renders App component inside ReduxProvider', () => {
    render(<AppProvider />);

    // Check if both wrapper and app are rendered
    expect(screen.getByTestId('mock-redux-wrapper')).toBeInTheDocument();
    expect(screen.getByTestId('mock-app')).toBeInTheDocument();
  });

  it('maintains correct component hierarchy', () => {
    render(<AppProvider />);

    // Check if App is nested inside ReduxProvider
    const wrapper = screen.getByTestId('mock-redux-wrapper');
    const app = screen.getByTestId('mock-app');
    expect(wrapper).toContainElement(app);
  });

  it('matches snapshot', () => {
    const { container } = render(<AppProvider />);
    expect(container).toMatchSnapshot();
  });

  it('renders App component exactly once', () => {
    render(<AppProvider />);
    expect(App).toHaveBeenCalledTimes(1);
  });

  it('provides Redux store to the app', () => {
    render(<AppProvider />);

    // Verify ReduxProvider was called with store
    expect(ReduxProvider).toHaveBeenCalledTimes(1);
  });

  it('has correct structure and content', () => {
    const { container } = render(<AppProvider />);

    // Check the DOM structure
    expect(container.firstChild).toBeInTheDocument();
    expect(screen.getByText('Mock App')).toBeInTheDocument();
  });
});
