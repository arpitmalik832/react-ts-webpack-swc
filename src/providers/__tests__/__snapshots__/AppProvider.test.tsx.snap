// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppProvider Tests matches snapshot 1`] = `
<div>
  <div
    data-testid="mock-redux-wrapper"
  >
    <div
      data-testid="mock-app"
    >
      Mock App
    </div>
  </div>
</div>
`;
